### PR TITLE
1705 Add unit tests to AuthenticatedURLSession

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

File: Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import XCTest
+import Cuckoo
+@testable import AccountSDKIOSWeb
+
+class XCTestCaseWithMockHTTPClient: XCTestCase {
```
Comment (Daniel-Echegaray) [inline Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift:5]: Nice!

File: Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    func stubHTTPClientExecuteRequest(mockHTTPClient: MockHTTPClient, result: Result<TestResponse, HTTPError>) {
+        stub(mockHTTPClient) { mock in
+            // makeRequest on saveRequestOnRefreshSuccess
+            when(mock.execute(request: any(), withRetryPolicy: any(), completion: anyClosure()))
+                .then { (_, _, completion: HTTPResultHandler<TestResponse>) in
+                    completion(result)
+                }
+        }
+    }
+    func stubSessionStorageStore(mockSessionStorage: MockSessionStorage, result: Result<Void, Error>) {
```
Comment (Daniel-Echegaray) [inline Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift:35]: new line

**Review by Daniel-Echegaray — APPROVED** [review]
lgtm!

File: Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/HTTP/AuthenticatedURLSessionTests.swift [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        let tokenResponse: TokenResponse = TokenResponse(access_token: "newAccessToken", refresh_token: "newRefreshToken", id_token: nil, scope: nil, expires_in: 3600)
+        self.stubHTTPClientExecuteRefreshRequest(mockHTTPClient: mockHTTPClient!, refreshResult: .success(tokenResponse))
+        
+        let mockSessionStorage = MockSessionStorage()
+        self.stubSessionStorageStore(mockSessionStorage: mockSessionStorage, result: .success())
+        
+        let client = Client(configuration: Fixtures.clientConfig, sessionStorage: mockSessionStorage, stateStorage: StateStorage(), httpClient: mockHTTPClient!)
+        let user = User(client: client, tokens: Fixtures.userTokens)
+        
+        let expectation = self.expectation(description: "Original request should be retired with success server response")
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/HTTP/AuthenticatedURLSessionTests.swift]: done

File: Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift (hunk 2 above)
Comment (zamorastp) [reply Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/XCTestCaseWithMockHTTPClient.swift:36]: Fixed

**Pull request merged**